### ISSUE TITLE
FrozenError: can't modify frozen String

### ISSUE THREAD

**Issue by MatUrbanski** [sferik/rails_admin#3139]
From time to time in my production environment `FrozenError: can't modify frozen String` error  is raised. Here is the backtrace:

```

rack-pjax-1.0.0/lib/rack/pjax.rb:19 → force_encoding
--
rack-pjax-1.0.0/lib/rack/pjax.rb:19 → block in call
actionpack-5.2.1/lib/action_dispatch/http/response.rb:147 → each
actionpack-5.2.1/lib/action_dispatch/http/response.rb:147 → each_chunk
actionpack-5.2.1/lib/action_dispatch/http/response.rb:128 → each
actionpack-5.2.1/lib/action_dispatch/http/response.rb:76 → each
actionpack-5.2.1/lib/action_dispatch/http/response.rb:475 → each
rack-pjax-1.0.0/lib/rack/pjax.rb:18 → call
rack-2.0.5/lib/rack/method_override.rb:22 → call

rack-2.0.5/lib/rack/session/abstract/id.rb:232 → context
rack-2.0.5/lib/rack/session/abstract/id.rb:226 → call
actionpack-5.2.1/lib/action_dispatch/middleware/cookies.rb:670 → call
rack-attack-5.4.2/lib/rack/attack.rb:182 → call
app/middleware/catch_json_parse_errors.rb:9 → call
remotipart-1.4.2/lib/remotipart/middleware.rb:32 → call
warden-1.2.7/lib/warden/manager.rb:36 → block in call
warden-1.2.7/lib/warden/manager.rb:35 → catch
warden-1.2.7/lib/warden/manager.rb:35 → call
rack-2.0.5/lib/rack/tempfile_reaper.rb:15 → call
rack-2.0.5/lib/rack/etag.rb:25 → call
rack-2.0.5/lib/rack/conditional_get.rb:25 → call
rack-2.0.5/lib/rack/head.rb:12 → call
actionpack-5.2.1/lib/action_dispatch/http/content_security_policy.rb:18 → call
rack-2.0.5/lib/rack/session/abstract/id.rb:232 → context
rack-2.0.5/lib/rack/session/abstract/id.rb:226 → call
actionpack-5.2.1/lib/action_dispatch/middleware/cookies.rb:670 → call
actionpack-5.2.1/lib/action_dispatch/middleware/callbacks.rb:28 → block in call
activesupport-5.2.1/lib/active_support/callbacks.rb:98 → run_callbacks
actionpack-5.2.1/lib/action_dispatch/middleware/callbacks.rb:26 → call
airbrake-6.1.2/lib/airbrake/rack/middleware.rb:52 → call
actionpack-5.2.1/lib/action_dispatch/middleware/debug_exceptions.rb:61 → call
actionpack-5.2.1/lib/action_dispatch/middleware/show_exceptions.rb:33 → call
railties-5.2.1/lib/rails/rack/logger.rb:38 → call_app
railties-5.2.1/lib/rails/rack/logger.rb:26 → block in call
activesupport-5.2.1/lib/active_support/tagged_logging.rb:71 → block in tagged
activesupport-5.2.1/lib/active_support/tagged_logging.rb:28 → tagged
activesupport-5.2.1/lib/active_support/tagged_logging.rb:71 → tagged
railties-5.2.1/lib/rails/rack/logger.rb:26 → call
actionpack-5.2.1/lib/action_dispatch/middleware/remote_ip.rb:81 → call
request_store-1.4.1/lib/request_store/middleware.rb:19 → call
actionpack-5.2.1/lib/action_dispatch/middleware/request_id.rb:27 → call
rack-2.0.5/lib/rack/method_override.rb:22 → call
rack-2.0.5/lib/rack/runtime.rb:22 → call
google-cloud-trace-0.33.2/lib/google/cloud/trace/middleware.rb:156 → block in call
google-cloud-trace-0.33.2/lib/google/cloud/trace.rb:441 → block in in_span
google-cloud-trace-0.33.2/lib/google/cloud/trace/trace_record.rb:232 → in_span
google-cloud-trace-0.33.2/lib/google/cloud/trace.rb:438 → in_span
google-cloud-trace-0.33.2/lib/google/cloud/trace/middleware.rb:154 → call
actionpack-5.2.1/lib/action_dispatch/middleware/executor.rb:14 → call
actionpack-5.2.1/lib/action_dispatch/middleware/static.rb:127 → call
rack-2.0.5/lib/rack/sendfile.rb:111 → call
rack-2.0.5/lib/rack/deflater.rb:34 → call
rack-cors-1.0.2/lib/rack/cors.rb:97 → call
railties-5.2.1/lib/rails/engine.rb:524 → call
/usr/lib/ruby/vendor_ruby/phusion_passenger/rack/thread_handler_extension.rb:97 → process_request

/usr/lib/ruby/vendor_ruby/phusion_passenger/request_handler/thread_handler.rb:149 → accept_and_process_next_request
/usr/lib/ruby/vendor_ruby/phusion_passenger/request_handler/thread_handler.rb:110 → main_loop
/usr/lib/ruby/vendor_ruby/phusion_passenger/request_handler.rb:415 → block (3 levels) in start_threads
/usr/lib/ruby/vendor_ruby/phusion_passenger/utils.rb:113 → block in create_thread_and_abort_on_exception


```

I'm using ruby 2.5.1p57 (2018-03-29 revision 63029) [x86_64-linux] and Ruby on Rails 5.2.1.

Is there any way do fix that?

**Comment by mshibuya** [comment]
I guess your app is returning something frozen. Using debugger will be useful to track down what's frozen.

**Comment by khadzhinov** [comment]
@mshibuya I had same issue with pjax and airbrake after airbrake upgrade. Seems that issue was fixed in pjax 1.1.0 (https://github.com/eval/rack-pjax/commit/75f3c29c4965b2be425a6875159bd940d2d52037) but I still have this issue
